simplq/src/components/pages/Admin/QueueHistory.jsx

```diff
@@ -1,13 +1,177 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import HistoryIcon from '@material-ui/icons/History';
 import SidePanelItem from 'components/common/SidePanel/SidePanelItem';
+import { useSelector } from 'react-redux';
+import { selectRemoveTokens, selectTokens } from 'store/selectedQueue';
+import moment from 'moment';
+import AddedInQueue from '../QueueHistory/AddedInQueue';
+import RemovedFromQueue from '../QueueHistory/RemovedFromQueue';
+import styles from './QueueHistory.module.scss';
 
 export default () => {
+  const [tokens, setTokens] = useState([]);
+  const [removedTokens, setRemovedTokens] = useState([]);
+  const [queueHistory, setQueueHistory] = useState([]);
+  const [showMore, setShowMore] = useState(false);
+  const [count, setCount] = useState(0);
+  const [sortedData, setSortedData] = useState([]);
+
+  const added = useSelector(selectTokens);
+  const removed = useSelector(selectRemoveTokens);
+
+  useEffect(() => {
+    (async () => {
+      let lastAdded = [];
+      let lastRemoved = [];
+      let lastRemovedPrime = [];
+      if (added) {
+        lastAdded = await added.filter((item) => {
+          let isPresent = false;
+          tokens.forEach((token) => {
+            if (item.tokenId === token.tokenId) {
+              isPresent = true;
+            }
+          });
+
+          return !isPresent;
+        });
+      }
```

Comment:
What's this code doing?

Reply:
As per the screenshot that I have attached, we can see that in removedTokens List we are not getting the timestamp of the new removed item, for this I am maintaining the queueHistory and moreover if we refresh we don't have anything in the queueHistory so we can't just add the first item that was newly added  which we could have done in all other cases.